modules/core/shared/src/main/scala/lucuma/core/model/Target.scala

```diff
@@ -63,7 +66,9 @@ object Target extends WithGid('t'.refined) with TargetOptics {
     name:          NonEmptyString,
     ephemerisKey:  EphemerisKey,
     sourceProfile: SourceProfile
-  ) extends Target
+  ) extends Target {
+    def at(i: Timestamp) = none
```

Comment:
I'm unfamiliar with how to get the `EphemerisCoords` out of `EphemerisKey`. Maybe this actual needs to be
`def at(i: Timestamp): F[TrackedCoordinates]` to take into account that they may come from a remote DB?

Reply:
I'm unfamiliar too, but if it's some kind of extrapolation to table values, I think it's safe to keep `Instant` in this interface.

Reply:
I assume we'll have a robot that keeps non-sidereal coordinates up-to-date like we do today.  I don't think the intention is to reach out to Horizons on demand.

Reply:
Just out of curiosity: how is it done today? Where does the robot store the non-sidereal coordinates?

Reply:
In the `ocs` codebase ODB, programs are mutable tree structures stored in memory (java-serialized to disk when modified).  Each non-sidereal observation references a relatively low-precision ephemeris table that covers the semester plus a month (or so) on either side. This is obtained from Horizons when the non-sidereal target is first resolved.

In addition, there is a robot that checks hourly for active observations, obtains high-precision ephemeris data for the upcoming or current observing night and stores them as text files in a place and format that the TCS can read and use for tracking.

Reply:
Do we have plans to do something like that in the odb? We'd update the ephemeris table and the odb will give us the current table on the graphql response?

Reply:
At least at the time the project was first coming together we suggested a target service separate from the ODB that would maintain this type of information.  Regardless I think ephemerides belong in a (relational) database table that is kept up-to-date automatically.  Whether to put them in graphql responses or transfer the data in some more efficient way (or take advantage of the client being able to make the same Horizons queries?) I don't know.  What do you guys think?

Reply:
I don't have a strong opinion. I guess engage will need to query the ephemeris eventually though again it could do it directly.
Do we need API users to get ephemeris?

Reply:
When is it that the non-sidereal target first resolved?

Does the high-precision ephemeris data come from Horizons too?

Could engage and explore query Horizons directly? Is there a requirement that resulted in the robot writing a text file instead of the TCS querying Horizons?